File: backend/app/schemas/pet.py
from pydantic import BaseModel, Field


class PetActionOut(BaseModel):
    key: str
    label: str
    icon: str = "✨"
    animation_row: int = 0
    frame_count: int = 6
    fps: int = 8
    loop: bool = False
    route: str = ""


class PetManifestOut(BaseModel):
    slug: str
    name: str
    description: str = ""
    preview_url: str = ""
    manifest_url: str = ""
    sprite_url: str = ""
    format: str = "spritesheet"
    columns: int = 1
    rows: int = 1
    cell_width: int = 0
    cell_height: int = 0
    sheet_width: int = 0
    sheet_height: int = 0
    animation_row: int = 0
    frame_count: int = 0
    fps: int = 12
    actions: list[PetActionOut] = []


class PetSelectRequest(BaseModel):
    pet_slug: str


class PetAffinityIn(BaseModel):
    delta: int = Field(default=1, ge=1, le=20)
    reason: str = ""


class PetAffinityOut(BaseModel):
    pet_affinity: int
    level: int
    level_name: str

File: frontend/src/api/pet.ts
import { apiGet, apiPost } from './client';

export interface PetAction {
  key: string;
  label: string;
  icon: string;
  animation_row: number;
  frame_count: number;
  fps: number;
  loop: boolean;
  route: string;
}

export interface PetManifest {
  slug: string;
  name: string;
  description: string;
  preview_url: string;
  manifest_url: string;
  sprite_url: string;
  format?: string;
  columns: number;
  rows: number;
  cell_width?: number;
  cell_height?: number;
  sheet_width?: number;
  sheet_height?: number;
  animation_row?: number;
  frame_count: number;
  fps: number;
  actions?: PetAction[];
}

export interface PetAffinity {
  pet_affinity: number;
  level: number;
  level_name: string;
}

export const fetchPets = () => apiGet<PetManifest[]>('/api/pets');
export const fetchOwnedPets = () => apiGet<{ owned: string[] }>('/api/pets/owned');
export const selectPet = (pet_slug: string) =>
  apiPost<{ pet_slug: string; pet_affinity?: number }>('/api/users/me/pet', { pet_slug });
export const bumpPetAffinity = (delta = 1, reason = '') =>
  apiPost<PetAffinity>('/api/pets/affinity', { delta, reason });
export const fetchPetAffinity = () => apiGet<PetAffinity>('/api/pets/affinity');

/** 免费桌宠（与后端 FREE_PET_SLUGS 保持一致） */
export const FREE_PET_SLUGS = ['boxcat', 'mallow', 'ghost', 'guami', 'pupu'] as const;
export const DEFAULT_PET_SLUG = 'boxcat';
